src/Amqp/Akka.Streams.Amqp.V1/AmqpSourceStage.cs

```diff
@@ -65,14 +68,37 @@ public AmqpSourceStageLogic(AmqpSourceStage<T> stage, Attributes attributes) : b
                 }, onDownstreamFinish: CompleteStage);
             }
 
+            private void HandleConnectionResult(ConnectResult result)
+            {
+                _connectionThread.Join(100);
+                _connectionThread = null;
+
+                if (!result.IsSuccessful)
+                {
+                    FailStage(result.Exception);
+                    return;
+                }
+                Log.Info("Connected to AMQP.V1 server.");
+            }
+
             public override void PreStart()
             {
                 base.PreStart();
 
-                Connect().Wait();
+                _connectionThread = new Thread(() =>
```

Comment:
You don't need separate thread - use `Task.Factory.StartNew (() => Connect(),
TaskCreationOptions.LongRunning);` which will spawn a separate thread for your computation for you. The only thing is that method executed this way should not be async (so you'd have to change Task.Delay to Thread.Sleep, which should not really be a problem, as this thing is running on a dedicated thread anyway).

Reply:
> You don't need to explicitly create separate thread - use Task.Factory.StartNew (() => Connect(), TaskCreationOptions.LongRunning); which will spawn a separate thread for your computation for you. 

Yeah I agree with this - that'd be better.

Reply:
Good idea, I'll change the implementation.